### PR TITLE
fix(Payment Configuration): add PaymentConfigurationTransformProvider…

### PR DESCRIPTION
**Pull request by lydiagray**
… to get paymentconfig for environment

### Description
Payment Summary was no longer working (couldn't get to the config in S3) so added a new IPaymentConfigurationTransformProvider, and updated the PaymentConfigurationCheck and PayService

Update the Unit tests

### Checklist
- [x] Code compiles correctly
- [x] Created tests for the new changes
- [x] All tests passing
- [x] Extended the README / documentation, if necessary

### CODE REVIEW

File: src/Providers/Transforms/PaymentConfiguration/S3PaymentConfigurationTransformDataProvider.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                using (StreamReader reader = new StreamReader(responseStream))
+                {
+                    var responseBody = reader.ReadToEnd();
+                    return JsonConvert.DeserializeObject<T>(responseBody);
+                }
+            }
+            catch (AmazonS3Exception e)
+            {
+                var ex = new Exception($"S3PaymentConfigurationTransformDataProvider: An error has occured while attempting to get S3 Object, Exception: {e.Message}. {_environment.EnvironmentName.ToS3EnvPrefix()}/payment-config/paymentconfiguration.{_environment.EnvironmentName.ToS3EnvPrefix()}.json ", e);
+                throw ex;
```
Comment (ColinLees) [inline src/Providers/Transforms/PaymentConfiguration/S3PaymentConfigurationTransformDataProvider.cs:42]: If we are throwing the exception here we don't need to assign it to a var first

File: src/Providers/Transforms/PaymentConfiguration/S3PaymentConfigurationTransformDataProvider.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            }
+            catch (Exception e)
+            {
+                var ex = new Exception($"S3PaymentConfigurationTransformDataProvider: An error has occured while attempting to deserialise object, Exception: {e.Message}", e);
+                throw ex;
```
Comment (ColinLees) [inline src/Providers/Transforms/PaymentConfiguration/S3PaymentConfigurationTransformDataProvider.cs:47]: If we are throwing the exception here we don't need to assign it to a var first

**Review by ColinLees — CHANGES_REQUESTED** [review]

File: src/Providers/Transforms/PaymentConfiguration/S3PaymentConfigurationTransformDataProvider.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        private readonly IWebHostEnvironment _environment;
+        private readonly IConfiguration _configuration;
+
+        public S3PaymentConfigurationTransformDataProvider(IS3Gateway s3Service, IWebHostEnvironment environment, IConfiguration configuration)
+        {
+            _s3Gateway = s3Service;
+            _environment = environment;
+            _configuration = configuration;
+        }
+
```
Comment (lallysmbc) [inline src/Providers/Transforms/PaymentConfiguration/S3PaymentConfigurationTransformDataProvider.cs:25]: can we add associated tests for this transform, dont need one for the local transform.

File: src/Services/PayService/PayService.cs [hunk 4]
```diff
@@ -108,7 +104,7 @@ public async Task<string> ProcessPaymentResponse(string form, string responseCod
 
         public async Task<PaymentInformation> GetFormPaymentInformation(MappingEntity formData, string form, Page page)
         {
-            var paymentConfig = await _cache.GetFromCacheOrDirectlyFromSchemaAsync<List<PaymentInformation>>($"paymentconfiguration.{_hostingEnvironment.EnvironmentName}", _distributedCacheExpirationConfiguration.PaymentConfiguration, ESchemaType.PaymentConfiguration);
+            var paymentConfig = await _paymentConfigProvider.Get<List<PaymentInformation>>();
             var formPaymentConfig = paymentConfig.FirstOrDefault(_ => _.FormName == form);
```
Comment (lallysmbc) [inline src/Services/PayService/PayService.cs:108]: can you just  check if _cache is used anywhere else in the project if not can be removed,i believe it is not anymore, might be wrong.

File: tests/unit-tests/UnitTests/Services/PayServiceTests.cs [hunk 5]
```diff
@@ -132,10 +121,8 @@ public PayServiceTests()
                 .ReturnsAsync(mappingEntity);
             _mockHostingEnvironment.Setup(_ => _.EnvironmentName).Returns("local");
 
-            _service = new PayService(_mockPaymentProvider.Object, _mockLogger.Object, _mockGateway.Object,
-                _mockCache.Object,
-                _mockDistributedCacheExpirationSettings.Object, _mockSessionHelper.Object, _mockMappingService.Object,
-                _mockHostingEnvironment.Object, _mockPageHelper.Object);
+            _service = new PayService(_mockPaymentProvider.Object, _mockLogger.Object, _mockGateway.Object, _mockSessionHelper.Object, _mockMappingService.Object,
+                _mockHostingEnvironment.Object, _mockPageHelper.Object, _mockPaymentConfigProvider.Object);
```
Comment (lallysmbc) [inline tests/unit-tests/UnitTests/Services/PayServiceTests.cs:125]: any test within here able to be updated with a verify call to the new provider?

File: src/Providers/Transforms/PaymentConfiguration/S3PaymentConfigurationTransformDataProvider.cs (hunk 1 above)
Comment (lydiagray) [reply src/Providers/Transforms/PaymentConfiguration/S3PaymentConfigurationTransformDataProvider.cs]: Amended, also replicated in the other Transforms

File: src/Services/PayService/PayService.cs (hunk 4 above)
Comment (lydiagray) [reply src/Services/PayService/PayService.cs:108]: Looks like it was used in PageHelper at one point, but now is unused. I'll remove it and the tests too

**Review by ColinLees — APPROVED** [review]

**Pull request merged**